src/lib/operators/sort.cpp

```diff
@@ -56,98 +56,140 @@ class Sort::SortImplMaterializeOutput {
 
   std::shared_ptr<const Table> execute() {
     // First we create a new table as the output
-    auto output = std::make_shared<Table>(_table_in->column_definitions(), TableType::Data, _output_chunk_size);
 
     // We have decided against duplicating MVCC data in https://github.com/hyrise/hyrise/issues/408
-
-    // After we created the output table and initialized the column structure, we can start adding values. Because the
-    // values are not ordered by input chunks anymore, we can't process them chunk by chunk. Instead the values are
-    // copied column by column for each output row. For each column in a row we visit the input segment with a reference
-    // to the output segment. This enables for the SortImplMaterializeOutput class to ignore the column types during the
-    // copying of the values.
     const auto row_count_out = _row_id_value_vector->size();
+    std::shared_ptr<Table> output;
+
+    if (_table_in->type() == TableType::Data) {
```

Comment:
Could you only include the necessary changes for the aggregate itself? I think the sort operator optimization should be a standalone PR.

Reply:
What's the status on the sort optimization PR?

Reply:
I am going to file the sort optimization PR. The aggregate PR should be unaffected.